lm_engine/modeling_utils/sequence_mixer_blocks/mamba2/op.py

```
@@ -0,0 +1,457 @@
+def mamba2_torch(
+    x: torch.Tensor,
+    dt: torch.Tensor,
+    A_neg: torch.Tensor,
+    B: torch.Tensor,
+    C: torch.Tensor,
+    D: torch.Tensor,
+    h: torch.Tensor | None,
+    use_recurrent: bool,
+    chunk_size: int,
+) -> tuple[torch.Tensor, torch.Tensor]:
+    BS, S, I = x.size()
```

Ruff (0.15.21)

[error] Ambiguous variable name: `I`

(E741)